File: src/main/java/com/gamyacouture/catalog/application/CategoryImages.java
package com.gamyacouture.catalog.application;

import org.springframework.util.StringUtils;

/**
 * Default cover / product placeholder URLs (Unsplash) keyed by category slug patterns.
 */
public final class CategoryImages {

    private static final String DEFAULT =
            "https://images.unsplash.com/photo-1583391734527-658aeeef0f35?w=1200&q=80";

    private CategoryImages() {
    }

    public static String coverForSlug(String slug) {
        if (!StringUtils.hasText(slug)) {
            return DEFAULT;
        }
        String s = slug.toLowerCase();
        if (s.contains("saree")) {
            return "https://images.unsplash.com/photo-1610030469983-98e550b19538?w=1200&q=80";
        }
        if (s.contains("lehenga") || "bridal".equals(s)) {
            return "https://images.unsplash.com/photo-1595777457583-95e059d581b8?w=1200&q=80";
        }
        if (s.contains("kurt")) {
            return "https://images.unsplash.com/photo-1617627143750-d86bc21e3273?w=1200&q=80";
        }
        if (s.contains("blouse")) {
            return "https://images.unsplash.com/photo-1572804013309-59a23b2e4c1f?w=1200&q=80";
        }
        if (s.contains("frock") || s.contains("girl") || s.contains("kid") || s.contains("birthday")) {
            return "https://images.unsplash.com/photo-1515488042361-ee00e8170dc8?w=1200&q=80";
        }
        if (s.contains("sherwani") || "men".equals(s)) {
            return "https://images.unsplash.com/photo-1620799140408-8747d1d90e59?w=1200&q=80";
        }
        return DEFAULT;
    }

    public static String productForSlug(String slug) {
        return coverForSlug(slug).replace("w=1200", "w=800");
    }

    public static String productForName(String productName) {
        if (!StringUtils.hasText(productName)) {
            return productForSlug(null);
        }
        String n = productName.toLowerCase();
        if (n.contains("saree")) {
            return productForSlug("sarees");
        }
        if (n.contains("lehenga")) {
            return productForSlug("lehengas");
        }
        if (n.contains("kurta") || n.contains("kurti")) {
            return productForSlug("kurtas");
        }
        if (n.contains("blouse")) {
            return productForSlug("blouses");
        }
        if (n.contains("frock") || n.contains("girl") || n.contains("kid")) {
            return productForSlug("frocks");
        }
        if (n.contains("sherwani")) {
            return productForSlug("sherwanis");
        }
        return productForSlug(null);
    }
}

File: frontend/src/lib/category-images.ts
/** Category-aware cover and product placeholder images. */

const IMAGES = {
  saree: "/brand/category-saree.jpg",
  lehenga: "/brand/category-lehenga.jpg",
  kurti: "/brand/hero-saree.jpg",
  blouse: "/brand/hero-saree.jpg",
  frock: "/brand/category-girls.jpg",
  default: "/brand/hero-saree.jpg",
} as const;

const REMOTE_PLACEHOLDER_HOSTS = ["images.unsplash.com", "source.unsplash.com"] as const;

function matchSlug(slug: string): keyof typeof IMAGES {
  const s = slug.toLowerCase();
  if (s.includes("saree")) return "saree";
  if (s.includes("lehenga") || s === "bridal" || s === "festive") return "lehenga";
  if (s.includes("kurt") || s.includes("blouse")) return "kurti";
  if (s.includes("frock") || s.includes("girl") || s.includes("birthday") || s.includes("kid"))
    return "frock";
  return "default";
}

function isRemotePlaceholder(url: string): boolean {
  try {
    const host = new URL(url).hostname;
    return REMOTE_PLACEHOLDER_HOSTS.some((placeholder) => host.includes(placeholder));
  } catch {
    return false;
  }
}

export function categoryCoverImage(slug: string, imageUrl?: string | null): string {
  const fallback = IMAGES[matchSlug(slug)];
  if (!imageUrl?.trim() || isRemotePlaceholder(imageUrl)) return fallback;
  return imageUrl;
}

export function productPlaceholderImage(
  primaryCategorySlug?: string | null,
  productName?: string | null,
): string {
  if (primaryCategorySlug) {
    return categoryCoverImage(primaryCategorySlug);
  }
  if (productName) {
    const n = productName.toLowerCase();
    if (n.includes("saree")) return IMAGES.saree;
    if (n.includes("lehenga")) return IMAGES.lehenga;
    if (n.includes("kurta") || n.includes("kurti") || n.includes("blouse")) return IMAGES.kurti;
    if (n.includes("frock") || n.includes("girl")) return IMAGES.frock;
  }
  return IMAGES.default;
}

export function normalizeProductImage(
  imageUrl: string | null | undefined,
  primaryCategorySlug?: string | null,
  productName?: string | null,
): string {
  if (!imageUrl?.trim() || isRemotePlaceholder(imageUrl)) {
    return productPlaceholderImage(primaryCategorySlug, productName);
  }
  return imageUrl;
}
